addons/mail/tests/test_mail_composer.py

```diff
@@ -72,3 +72,24 @@ def test_mail_mass_mode_compose_with_mso(self):
         self.assertIn(self.body_html,
             values[self.partner_employee.id]['body_html'],
             'We must preserve (mso) comments in email html')
+
+    def test_mail_mass_mode_compose_with_multiple_emails(self):
+        test_user = mail_new_test_user(self.env, login="test_user_emails", groups="base.group_user", email="testmail1@test.com,testmail2@test.com")
+        test_partner = test_user.partner_id
+        composer = self.env['mail.compose.message'].with_context({
+            'default_model': self.test_record._name,
+            'default_composition_mode': 'mass_mail',
+            'active_ids': [self.test_record.id],
+            'active_model': self.test_record._name,
+            'active_id': self.test_record.id
+        }).create({
+            'body': self.body_html,
+            'partner_ids': [(4, test_partner.id)],
+            'composition_mode': 'mass_mail',
+        })
+        with self.mock_mail_gateway(mail_unlink_sent=True):
+            composer._action_send_mail()
+
+        values = composer.get_mail_values(test_partner.ids)
```

Comment:
I would not do a low-level check on get mail values, but instead globally check the result of the send action. You may use ``assertMailMail`` tool method available in mail mock tools. You can grep it a bit to see how it is used, but idea is that when used with ``mock_mail_gateway``, sent emails are kept and we can check what is effectively send.